pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -164,16 +166,36 @@ func Execute(configFile io.Reader) {
 		log.Fatal(err)
 	}
 
-	if dockerConfig.HTTP.TLS.Certificate == "" {
-		dcontext.GetLogger(ctx).Infof("listening on %s", srv.Addr)
-		err = srv.ListenAndServe()
-	} else {
+	errc := make(chan error, 1)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		if dockerConfig.HTTP.TLS.Certificate == "" {
+			dcontext.GetLogger(ctx).Infof("listening on %s", srv.Addr)
+			errc <- srv.ListenAndServe()
+			return
+		}
+
 		dcontext.GetLogger(ctx).Infof("listening on %s, tls", srv.Addr)
-		err = srv.ListenAndServeTLS(dockerConfig.HTTP.TLS.Certificate, dockerConfig.HTTP.TLS.Key)
+		errc <- srv.ListenAndServeTLS(dockerConfig.HTTP.TLS.Certificate, dockerConfig.HTTP.TLS.Key)
+	}()
+
+	select {
+	case <-sigs:
+	case err := <-errc:
+		log.Fatal(err)
 	}
-	if err != nil {
+
+	// signal has been received, attempt to stop the server within the
+	// default termination grace period of 30 seconds.
+	ctx, cancel := context.WithTimeout(ctx, 29*time.Second)
```

Comment:
> period of 30 seconds

> `29*time.Second`

Why are the numbers different?

Reply:
The grace period is 30 seconds, i was planning to give it 1 entire second between receiving the signal and calling Shutdown(), maybe i could update the comment, what do you think?

Reply:
Yes, it worth documenting this difference so that the next reader won't be surprised.

Reply:
done.